mantidproject/mantid
# Refactoring and bug fixes for DPW and Refl GUI

This PR adds the following fixes and under-the-hood changes:
- Refactor GenericDataProcessorPresenter::reduceRow into a much shorter, more readable function
- Refactor ReflSettingsPresenter functions to return options as a map rather than constructing a csv string from them which then has to be re-parsed
- Consolidate csv parsing functions so that all components use the functions in ParseKeyValueString.h 
- Fix some issues from #21167:
  - Transmission runs entered on the Settings tab are not found #21076
  - You should be able to specify the transmission run on the settings tab or runs tab as a run number or a workspace name in the ADS - check that this still works.
  - Properties on the Runs tab should take precedence over properties on the Settings tab
- Improve naming of preprocessed output workspaces to preserve "+" in rather than replacing it with "_"

**To test:**

- Open `ISIS Reflectometry`
- Under `Run(s)`, enter `13460` on one row and add another row with `13461+13462`
- On the Settings tab enter the transmission runs = `13463,13464`
- Back on the Runs tab, select the group and hit Process
- Check that output workspaces include `IvsQ_13460_13461+13462`, `TRANS_13463+13464`, `TOF_13460` and `TOF_13461+13462`
- Delete all workspaces except these. Rename the `IvsQ_` workspace to e.g. `original` so that it doesn't get overwritten.
- Repeat with transmission runs = `"TRANS_13436+13464"` (including the quotes) and replacing the run numbers with `TOF_13460` (no quotes) and `"TOF_13461+13462"` (including quotes).
- You should get the same results as before except that `IvsQ_` is now named `IvsQ_TOF_` etc. Run `CompareWorkspaces` against `original` to check.
- There should be no workspaces named `TOF_` or `TRANS_` other than the ones used (i.e. no more workspaces were loaded).

Also check that the bug described in #21076 is fixed

Also open `ISIS SANS v2 experimental` and check that it still works by loading the mask and batch files from the sample data and processing them

Fixes #21218 and fixes #21076 

**Release Notes** 
*Added to the release notes*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
